components/checkbox/src/index.tsx

```diff
@@ -112,9 +133,10 @@ Checkbox.displayName = 'Checkbox';
 Checkbox.defaultProps = {
   hint: undefined,
   className: undefined,
+  size: 'medium',
 };
 
-export interface CheckboxProps extends React.InputHTMLAttributes<HTMLInputElement> {
+export interface CheckboxProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size'> {
```

Comment:
Are there any downsides to preventing the user from accessing the html input size prop?

Would it be better to use a name for our size prop that didn't clash, such as `sizeVariant`?